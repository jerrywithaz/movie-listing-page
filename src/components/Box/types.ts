export type BoxProps = {
    width: string,
    height: string
} & React.HTMLAttributes<HTMLDivElement>;