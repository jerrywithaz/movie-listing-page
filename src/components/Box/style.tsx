import React from 'react';
import styled from 'styled-components';
import { BoxProps } from './types';

export const Box = styled(({
    width, 
    height, 
    ...rest
}: BoxProps) => 
    <div {...rest}/>
)`
    height: ${props => props.height};
    width: ${props => props.width};
    box-shadow: ${props => props.theme.boxShadow.small};
    border-radius: ${props => props.theme.borderRadius.default}px;
    padding: ${props => props.theme.spacing.large}px;
`;