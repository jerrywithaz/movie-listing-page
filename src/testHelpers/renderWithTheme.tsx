import React from 'react';
import renderer from 'react-test-renderer';
import ThemeProvider from 'containers/ThemeProvider';

function renderWithTheme(component: JSX.Element) {
    return renderer.create(
        <ThemeProvider>
          {component}
        </ThemeProvider>
      );
}

export default renderWithTheme;