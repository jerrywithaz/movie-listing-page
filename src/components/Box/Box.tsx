import React from 'react';
import { BoxProps } from './types';

import * as Styled from './style';

const Box = (props: BoxProps) => {
    return (
        <Styled.Box {...props}/>
    );
}

export default Box;