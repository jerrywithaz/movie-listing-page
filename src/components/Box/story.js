import React from 'react';
import { storiesOf } from '@storybook/react';
import Box from './';

const story = storiesOf('Box', module);

story.add('Default', () => (
    <Box width="400px" height="400px">
        <div>hey</div>
    </Box>
));
  