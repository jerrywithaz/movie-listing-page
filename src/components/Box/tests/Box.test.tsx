import React from 'react';
import Box from './../';
import renderWithTheme from 'testHelpers/renderWithTheme';

describe("components/Box", () => {

    it("should match snapshot", () => {

        const tree = renderWithTheme(<Box width="400px" height="400px"/>).toJSON();

        expect(tree).toMatchSnapshot();

    });

});