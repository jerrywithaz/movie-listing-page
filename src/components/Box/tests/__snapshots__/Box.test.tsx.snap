// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Box should match snapshot 1`] = `
.c0 {
  height: 400px;
  width: 400px;
  box-shadow: 0px 1px 5px rgba(24,17,3,0.2), 0px 3px 4px rgba(0,0,0,0.12), 0px 2px 4px rgba(24,17,3,0.14);
  border-radius: 4px;
  padding: 32px;
  background: #ffffff;
}

<main
  className="c0"
/>
`;
